src/executioners/SpectralExecutionerLinearElastic.C

```diff
@@ -0,0 +1,402 @@
+/**********************************************************************/
+/*                     DO NOT MODIFY THIS HEADER                      */
+/* MAGPIE - Mesoscale Atomistic Glue Program for Integrated Execution */
+/*                                                                    */
+/*            Copyright 2017 Battelle Energy Alliance, LLC            */
+/*                        ALL RIGHTS RESERVED                         */
+/**********************************************************************/
+
+#include "SpectralExecutionerLinearElastic.h"
+#include "InputParameters.h"
+#include "RankFourTensor.h"
+
+registerMooseObject("MagpieApp", SpectralExecutionerLinearElastic);
+
+Real
+deltaij(const int i, const int j)
+{
+  if (i == j)
+    return 1.0;
+  else
+    return 0.0;
+}
+
+InputParameters
+SpectralExecutionerLinearElastic::validParams()
+{
+  InputParameters params = SpectralExecutionerBase::validParams();
+  params.addClassDescription("Executioner for spectral solve of diffusion equation");
+  params.addParam<Real>("time_step", 1.0, "Time step for ODE integration");
+  params.addParam<unsigned int>("number_iterations", 0, "Maximum number of iterations for solver");
+  params.addParam<Real>("young_modulus", 1.0, "First parameter for isotropic materials");
+  params.addParam<Real>("poisson_ratio", 1.0, "Second parameter for isotropic materials");
+  params.addParam<Real>("average_material_factor", 1.0, "Homogeneized factor for multiphase");
+  params.addRequiredParam<std::vector<Real>>(
+      "global_strain_tensor",
+      "Vector of values defining the constant applied global strain "
+      "to add. Components are XX, YY, ZZ, YZ, XZ, XY");
+  params.addParam<Real>("solver_error", 1.0e-4, "Error for fixed iteration solver");
+
+  return params;
+}
+
+SpectralExecutionerLinearElastic::SpectralExecutionerLinearElastic(
+    const InputParameters & parameters)
+  : SpectralExecutionerBase(parameters),
+    _dt(getParam<Real>("time_step")),
+    _nsteps(getParam<unsigned int>("number_iterations")),
+    _young_modulus(getParam<Real>("young_modulus")),
+    _poisson_ratio(getParam<Real>("poisson_ratio")),
+    _average_factor(getParam<Real>("average_material_factor")),
+    _solver_error(getParam<Real>("solver_error"))
+{
+
+  _initial_strain_tensor.fillFromInputVector(getParam<std::vector<Real>>("global_strain_tensor"));
+
+  _t_current = 0.0;
+}
+
+void
+SpectralExecutionerLinearElastic::fillOutEpsilonBuffer(
+    FFTBufferBase<RankTwoTensor> & epsilon_buffer)
+{
+  const auto & grid = epsilon_buffer.grid();
+  const int ni = grid[0];
+  const int nj = grid[1];
+  const int nk = grid[2];
+
+  std::size_t index = 0;
+
+  for (int freq_x = 0; freq_x < ni; ++freq_x)
+    for (int freq_y = 0; freq_y < nj; ++freq_y)
+      for (int freq_z = 0; freq_z < nk; ++freq_z)
+      {
+        epsilon_buffer.realSpace()[index] = _initial_strain_tensor;
+        index++;
+      }
+}
```

Comment:
how about we add

```
FFTData<T> & operator=(const T & rhs);
```

which internally uses `std::fill` to set the data. This will be more compact and easier to optimize for the compiler.

Reply:
The code in this file would then just be

```
epsilon_buffer.realSpace() = _initial_strain_tensor;
```

Reply:
Makes it much cleaner. The meaning of the operation becomes more implicit at first sight though.